app/src/test/java/io/github/droidkaigi/confsched2017/repository/contributors/ContributorsRepositoryTest.kt

```diff
@@ -0,0 +1,63 @@
+package io.github.droidkaigi.confsched2017.repository.contributors
+
+import io.github.droidkaigi.confsched2017.model.Contributor
+import io.reactivex.Single
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class ContributorsRepositoryTest {
+
+    @Test
+    @Throws(Exception::class)
+    fun findAllFromEmptyRepository() {
+        val localRepository = object : ContributorsReadWriteDataSource {
+            override fun findAll(): Single<List<Contributor>> = Single.just(listOf())
+
+            override fun updateAllAsync(contributors: List<Contributor>?) = Unit
+        }
+        val remoteRepository = object : ContributorsReadDataSource {
+            override fun findAll(): Single<List<Contributor>> = Single.just(listOf())
+        }
```

Comment:
Thanks again!

But I have a question.

I make DataSources `open public class` and use `mock<ContributorsLocalDataSource>()`.
But Mockito rises a reflection error on my Mac (not on Circle CI, it looks like there are no errors for me). 


```
org.mockito.exceptions.base.MockitoException: 
Mockito cannot mock this class: class io.github.droidkaigi.confsched2017.repository.contributors.ContributorsLocalDataSource.

Mockito can only mock non-private & non-final classes.
If you're not sure why you're getting this error, please report to the mailing list.


Java               : 1.8
JVM vendor name    : Oracle Corporation
JVM vendor version : 25.5-b02
JVM name           : Java HotSpot(TM) 64-Bit Server VM
JVM version        : 1.8.0_05-b13
JVM info           : mixed mode
OS name            : Mac OS X
OS version         : 10.12.3


Underlying exception : java.lang.IllegalArgumentException: object is not an instance of declaring class
```

https://github.com/kikuchy/conference-app-2017/blob/d26b8154e65baa1068c56c892b04516cad8f2357/app/src/test/java/io/github/droidkaigi/confsched2017/repository/contributors/ContributorsRepositoryTest.kt#L13

What is difference between your code and my code...? 🤔